### PR TITLE
vam: Add support for conditionals (case expr)

### PR DESCRIPTION
**Pull request by mattnibs**

### CODE REVIEW

File: runtime/vam/expr/conditional.go [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+				}
+			}
+		}
+	case *vector.Error:
+		if ridx != nil {
+			errs.AddMany(ridx)
+		} else {
+			errs.AddRange(0, uint64(vec.Len()))
+		}
+	}
```
Comment (nwt) [inline runtime/vam/expr/conditional.go:124]: Is the absence of a default case here intentional?

File: runtime/vam/op/filter.go [hunk 2]
```diff
@@ -31,18 +31,13 @@ func (f *Filter) Pull(done bool) (vector.Any, error) {
...
+	b, _ := expr.BoolMask(mask)
+	if b.IsEmpty() {
 		return nil, false
 	}
-	if len(index) == int(n) {
+	if b.GetCardinality() == uint64(mask.Len()) {
 		return vec, true
 	}
-	return vector.NewView(vec, index), true
+	return vector.NewView(vec, b.ToArray()), true
```
Comment (nwt) [inline runtime/vam/op/filter.go:42]: What does this change have to do with implementing conditionals?

Comment (mattnibs) [reply runtime/vam/op/filter.go:42]: Just figured not to have two different boolMask functions plus the new one is more optimized

File: runtime/vam/expr/conditional.go (hunk 1 above)
Comment (mattnibs) [reply runtime/vam/expr/conditional.go:124]: 👍🏻 should panic

Comment (mattnibs) [reply runtime/vam/expr/conditional.go:126]: On second thought I don't think it should. Have no default just sets the return value to false for non-bool vectors just like vector.BoolValue does

**Review by nwt — APPROVED** [review]

**Pull request merged**